.test-infra/jenkins/CommonTestProperties.groovy

```diff
@@ -16,20 +16,44 @@
  * limitations under the License.
  */
 
+
+
 class CommonTestProperties {
+    enum SDK {
+        PYTHON,
+        JAVA
+    }
+
     enum Runner {
-        DATAFLOW("DataflowRunner", ":beam-runners-google-cloud-dataflow-java"),
-        SPARK("SparkRunner", ":beam-runners-spark"),
-        FLINK("TestFlinkRunner", ":beam-runners-flink_2.11"),
-        DIRECT("DirectRunner", ":beam-runners-direct-java")
+        DATAFLOW("DataflowRunner"),
+        SPARK("SparkRunner"),
+        FLINK("TestFlinkRunner"),
+        DIRECT("DirectRunner"),
+
+        def RUNNER_DEPENDENCY_MAP = [
+                JAVA: [
+                        DATAFLOW: ":beam-runners-google-cloud-dataflow-java",
+                        SPARK: ":beam-runners-spark",
+                        FLINK: ":beam-runners-flink_2.11",
+                        DIRECT: ":beam-runners-direct-java"
+                ],
+                PYTHON: [
+                        DATAFLOW: "TestDataflowRunner",
+                        DIRECT: "DirectRunner",
+                ]
+        ]
 
         private final String option
-        private final String dependency
 
-        Runner(String option, String dependency) {
+        Runner(String option) {
             this.option = option
-            this.dependency = dependency
         }
+
+
+        String getDepenedencyBySDK(SDK sdk) {
+            RUNNER_DEPENDENCY_MAP.get(sdk.toString()).get(this.toString())
+        }
```

Comment:
@lgajowy This is a new implementation using SDK enum to get right dependency from map. I am no sure only if it's going to be clear for this enum user that you can use method of an enum.